core/src/services/alluxio/core.rs

```diff
@@ -0,0 +1,407 @@
+        let resp = self.client.send(req).await?;
+
+        let status = resp.status();
+        let body = resp.into_body().bytes().await?;
+
+        if status.is_success() {
+            return Ok(());
+        }
+        let err_response: ErrRsponse =
+            serde_json::from_slice(&body).map_err(new_json_serialize_error)?;
+        Err(err_response.into())
+    }
+
+    pub(super) async fn rename(&self, path: &str, dst: &str) -> Result<()> {
+        let op = Op::Rename;
+
+        let req = Request::post(format!(
+            "{}?dst=/{}",
+            self.join_path(path, op.as_str()),
+            dst
+        ));
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        let resp = self.client.send(req).await?;
+
+        let status = resp.status();
+        let body = resp.into_body().bytes().await?;
+
+        if status.is_success() {
+            return Ok(());
+        }
+        let err_response: ErrRsponse =
+            serde_json::from_slice(&body).map_err(new_json_serialize_error)?;
+        Err(err_response.into())
+    }
+
+    pub(super) async fn get_status(&self, path: &str) -> Result<FileInfo> {
+        let op = Op::GetStatus;
+
+        let req = Request::post(self.join_path(path, op.as_str()));
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        let resp = self.client.send(req).await?;
+
+        let status = resp.status();
+        let body = resp.into_body().bytes().await?;
+
+        if status.is_success() {
+            let file_info: FileInfo =
+                serde_json::from_slice(&body).map_err(new_json_deserialize_error)?;
+            return Ok(file_info);
+        }
+        let err_response: ErrRsponse =
+            serde_json::from_slice(&body).map_err(new_json_serialize_error)?;
+        Err(err_response.into())
+    }
+
+    pub(super) async fn list_status(&self, path: &str) -> Result<Vec<FileInfo>> {
+        let op = Op::ListStatus;
+
+        let req = Request::post(self.join_path(path, op.as_str()));
+
+        let req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+
+        let resp = self.client.send(req).await?;
+
+        let status = resp.status();
+        let body = resp.into_body().bytes().await?;
+
+        if status.is_success() {
+            let file_infos: Vec<FileInfo> =
+                serde_json::from_slice(&body).map_err(new_json_deserialize_error)?;
+            return Ok(file_infos);
+        }
+        let err_response: ErrRsponse =
+            serde_json::from_slice(&body).map_err(new_json_serialize_error)?;
+        Err(err_response.into())
+    }
+
+    pub async fn read(
+        &self,
+        stream_id: u64,
+        range: BytesRange,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let op = Op::Read;
+
+        let mut req = Request::post(self.join_stream(stream_id, op.as_str()));
```

Comment:
Maybe we can return the stream id in reader directly so that we don't need to `open_file` every time users calling `read`?

We can treat this as a future improvement.